check_log_ng.py

```diff
@@ -61,38 +61,40 @@ class LogChecker(object):
         {'_PERCENT_': '%'},
     ]
 
-    def __init__(self, initial_data):
+    def __init__(self, config):
         """ Constructor."""
         # set default value
-        self.logformat = LogChecker.FORMAT_SYSLOG
-        self.pattern_list = []
-        self.critical_pattern_list = []
-        self.negpattern_list = []
-        self.critical_negpattern_list = []
-        self.case_insensitive = False
-        self.encoding = 'utf-8'
-        self.warning = 1
-        self.critical = 0
-        self.nodiff_warn = False
-        self.nodiff_crit = False
-        self.trace_inode = False
-        self.multiline = False
-        self.scantime = 86400
-        self.expiration = 691200
-        self.cache = False
-        self.cachetime = 60
-        self.lock_timeout = 3
-
-        # set initial_data
-        for key in initial_data:
-            setattr(self, key, initial_data[key])
+        self.config = {}
+        self.config['logformat'] = LogChecker.FORMAT_SYSLOG
+        self.config['pattern_list'] = []
+        self.config['critical_pattern_list'] = []
+        self.config['negpattern_list'] = []
+        self.config['critical_negpattern_list'] = []
+        self.config['case_insensitive'] = False
+        self.config['encoding'] = 'utf-8'
+        self.config['warning'] = 1
+        self.config['critical'] = 0
+        self.config['nodiff_warn'] = False
+        self.config['nodiff_crit'] = False
+        self.config['trace_inode'] = False
+        self.config['multiline'] = False
+        self.config['scantime'] = 86400
+        self.config['expiration'] = 691200
+        self.config['cache'] = False
+        self.config['cachetime'] = 60
+        self.config['lock_timeout'] = 3
+
+        # overwrite values with user's values
+        for key in self.config:
+            if key in config and config[key]:
```

Comment:
I think `and config[key]` is too sensitive.
(Just now, there are not but effect, but I think, this behavior is not good for future.)

If in case value(`config[key`) is `0` (int 0) or `""` (blank str) or `[]` (empty list), config will not overwrite.

Reply:
Thank you for your comment.
I fix it.

Reply:
Thanks. looks good!